src/components/UpsertMovieModal.js

```diff
@@ -0,0 +1,135 @@
+import "./UpsertMovieModal.css";
+import React, {useContext} from "react";
+import { useForm } from "react-hook-form";
+import MoviesContext, { MovieRepo } from "../services/MoviesContext";
+import GenresRepo from "../services/GenreRepo";
+
+
+export default function UpsertMovieModal({header, movie, handleCloseModal}){
+    let contextValue = useContext(MoviesContext);
+    const genreRepo = new GenresRepo();
+    const genres = genreRepo.getAllGenres();
+
+    const movieRepo = contextValue.movieRepo;
+     
+    const { register, handleSubmit } = useForm();
+    const onSubmit = (data, e) => {
+        if (movie === undefined){
+            handleCreate(data);
+        } else {
+            handleUpdate(movie, data);
+        }
+
+        console.log(contextValue);
+
+        contextValue.setMovieRepo(movieRepo.clone());
+        handleCloseModal();
+    };
+    const onError = (errors, e) => console.log(errors, e);
+
+    const handleUpdate = (movie, formData)=>{
+        if(formData.title && formData.title !==""){
```

Comment:
you could just leave `formData.title`, in JS empty string in condition will convert to false

Reply:
done